src/OrchardCore/OrchardCore.Redis.Abstractions/RedisDatabaseFactory.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using StackExchange.Redis;
+
+namespace OrchardCore.Redis
+{
+    /// <summary>
+    /// Host level factory allowing to share a <see cref="IDatabase"/> across tenants.
+    /// </summary>
+    public sealed class RedisDatabaseFactory : IRedisDatabaseFactory, IDisposable, IAsyncDisposable
```

Comment:
This should not be in Redis.Abstractions. I understand why it's there, and the OC.Infra project should not depend on Redis.